samples/bluetooth/direct_test_mode/src/main.c

```diff
@@ -4,98 +4,30 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
-#include <errno.h>
-#include <zephyr/drivers/uart.h>
-#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/printk.h>
 
-#include "dtm.h"
+#include "transport/dtm_uart_twowire.h"
 
 int main(void)
 {
 	int err;
-	const struct device *uart = DEVICE_DT_GET(DT_CHOSEN(ncs_dtm_uart));
-	bool is_msb_read = false;
-	uint8_t rx_byte;
-	uint16_t dtm_cmd;
-	uint16_t dtm_evt;
-	int64_t msb_time;
+	uint16_t cmd;
 
 	printk("Starting Direct Test Mode example\n");
 
-	if (!device_is_ready(uart)) {
-		printk("UART device not ready\n");
-	}
-
-	err = dtm_init();
+	err = dtm_uart_twowire_init();
 	if (err) {
-		printk("Error during DTM initialization: %d\n", err);
+		printk("Error initializing DTM Two Wire Uart transport: %d\n", err);
+		return err;
 	}
 
 	for (;;) {
-		dtm_wait();
-
-		err = uart_poll_in(uart, &rx_byte);
+		cmd = dtm_uart_twowire_get();
+		err = dtm_uart_twowire_process(cmd);
```

Comment:
The name should be specific, because the functions could have different return and input parameters with different transport.
So the `main.c file` would serve as an abstraction for the different transports.